engine/src/main/java/org/apache/apex/engine/plugin/ApexPluginManager.java

```diff
@@ -187,5 +189,19 @@ PluginInfo getPluginInfo(DAGExecutionPlugin plugin)
     {
       ApexPluginManager.this.register(type, handler, owner);
     }
+
+    @Override
+    public DAG getDAG()
+    {
+      return clonedDAG.get();
+    }
+  }
+
+  @Override
+  public <T> void dispatch(RegistrationType<T> registrationType, T data)
+  {
+    if (registrationType == ApexPluginDispatcher.DAG_CHANGE_EVENT) {
+      clonedDAG.set(SerializationUtils.clone((DAG)data));
```

Comment:
This will be called only when dag is changed and during Stram startup, and cloned dag is kept for future reference when user plugin has asked for it.